JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/favourites/FavouriteScreenViewModel.kt

```diff
@@ -16,33 +16,99 @@
 
 package com.google.jetstream.presentation.screens.favourites
 
+import androidx.compose.runtime.Immutable
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.google.jetstream.data.entities.MovieList
 import com.google.jetstream.data.repositories.MovieRepository
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.SharingStarted
-import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class FavouriteScreenViewModel @Inject constructor(
     movieRepository: MovieRepository
 ) : ViewModel() {
 
-    val uiState = movieRepository.getFavouriteMovies().map {
-        FavouriteScreenUiState.Ready(it)
+    private val selectedFilterListFlow: MutableSharedFlow<FilterList> = MutableSharedFlow()
+
+    private val selectedFilterList = selectedFilterListFlow.stateIn(
+        viewModelScope,
+        SharingStarted.WhileSubscribed(5_000),
+        FilterList()
+    )
```

Comment:
Why `MutableSharedFlow`, which calls `stateIn`? Can you use `MutableStateFlow(FilterList())` directly?

Reply:
Thanks. I did not realize that we can set values to MutableStateFlow. Updated the code as suggested.